pkg/cmd/server/start/start_master.go

```diff
@@ -429,7 +429,7 @@ func StartAPI(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) error {
 	}
 
 	// verify we can connect to etcd with the provided config
-	if err := etcd.TestEtcdClient(oc.EtcdClient); err != nil {
+	if _, err := etcd.GetAndTestEtcdClient(oc.Options.EtcdClientInfo); err != nil {
```

Comment:
I thought that the TestEtcdClient is used here only to be locked up until etcd is not available.

I can do this check as a method in MasterConfig and make EtcdClient private.

Cc: @pweil-